applications/nrf_desktop/src/modules/hid_forward.c

```diff
@@ -880,7 +880,7 @@ static void disconnect_peripheral(struct hids_peripheral *per)
 	__ASSERT_NO_MSG(!is_any_report_enqueued(&per->enqueued_reports));
 
 	bt_hogp_release(&per->hogp);
-	k_delayed_work_cancel(&per->read_rsp);
```

Comment:
There is a potential race that will need to be carefully reviewed. The problem will happen if cancel will block resubmission done in the other thread. We need to check how these callbacks are called.